lib/iris/plot.py

```diff
@@ -403,8 +405,14 @@ def _draw_2d_from_points(draw_method_name, arg_func, cube, *args, **kwargs):
 
 def _fixup_dates(coord, values):
     if coord.units.calendar is not None and values.ndim == 1:
-        r = [datetime.datetime(*(coord.units.num2date(val).timetuple()[0:6]))
-             for val in values]
+        num2date = coord.units.num2date
+        if coord.units.calendar == 'gregorian':
+            r = [datetime.datetime(*(num2date(val).timetuple()[0:6]))
```

Comment:
I have factored out `num2date(val).timetuple()[0:6]` but I have to leave the unpacking until the datetime call as each of the 6 values in the tuple are arguments of the datetime.datetime or netcdftime.datetime
i.e.
```
>>> d = (coord.units.num2date(2).timetuple()[0:6])
>>> print d
(2000, 2, 3, 0, 0, 0)
>>> print datetime.datetime(*d)
2000-02-03 00:00:00

```